core/src/main/java/net/opentsdb/query/filter/AnyFieldRegexFilter.java

```diff
@@ -95,6 +98,38 @@ public String toString() {
             .toString();
   }
 
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o)
+      return true;
+    if (o == null || getClass() != o.getClass())
+      return false;
+
+    final AnyFieldRegexFilter otherRegexFilter = (AnyFieldRegexFilter) o;
+
+
+    return Objects.equal(filter, otherRegexFilter.getFilter())
+            && Objects.equal(pattern.pattern(), otherRegexFilter.pattern().pattern());
+
+  }
+
+
+  @Override
+  public int hashCode() {
+    return buildHashCode().asInt();
+  }
+
+
+  /** @return A HashCode object for deterministic, non-secure hashing */
+  public HashCode buildHashCode() {
+    final HashCode hc = Const.HASH_FUNCTION().newHasher()
+            .putString(Strings.nullToEmpty(filter), Const.UTF8_CHARSET)
+            .putString(Strings.nullToEmpty(pattern.pattern()), Const.UTF8_CHARSET)
```

Comment:
This (And other filters) need their type as well. The reason being we could have a `TagValueRegexp` with a filter like `.*` and a `FieldRegexp` filter with the same filter and want them to hash to something different.

For this and other regex filters, you can omit the pattern from equals and hash as those are just compiled versions of the `filter` field. It's ok not to use the parsed version even if someone puts a space in front or back.